### DISCUSSION TITLE
portal mode

### DISCUSSION THREAD

**Discussion by fables-tales** [BattlesnakeOfficial/feedback#120] [General]
* snakes lose 1hp per turn as usual
* there are always exactly two foods on the board
* colliding a food resets you to 100 and puts your head at the other food, and removes both foods, and two new foods spawn
* your tail has to go all the way through the portal before your body continues to move as normal
* your head comes out the portal in the same direction it was moving previously

**Comment by fables-tales** [comment]
stolen shamelessly from snake as implemented on google.com (punch "snake game" in to the search)

**Comment by gcr** [comment]
I love this idea but as stated, this might be either uninteresting or underspecified.

On the turn that the snake head collects a food and appears on the other portal, is the snake allowed to proceed in any of the four unopposed directions? That appears to contradict your last point since the snake is free to turn in any direction.

On the other hand, if the snake must continue straight until their tail exits the portal, snakes that have greater length than the length of the board must hit a wall, which feels like a harsh limitation. 

**Proposed alternative:** snakes may not change direction for one move after the head enters the wormhole, replacing your rules 4 and 5. This still requires snakes to plan ahead a bit without sending long snakes into the wall.

**Reply by fables-tales** [reply]
I think the idea might be that if the snake is in the "out" portal square, that it must move at least once in the same direction it was previously going when it entered the portal, or it dies, but after that is free to move as it wants

**Reply by gcr** [reply]
Ah we both have the same idea (see my addendum) :p

**Comment by robbles** [comment]
I think this is an awesome idea :)

I'm not 100% sure I understand this bit:
>your tail has to go all the way through the portal before your body continues to move as normal

Is that just saying that your snake body acts as if it's continuous through the portal, as opposed to forming a new "tail" after a few moves?

**Reply by InterestingBrainPoops** [reply]
I think it could be generalized as follows: 
When you eat a food, your head positon becomes the second foods location. And you add 2 tails on top of your old tail, and then you proceed as follows. Otherwise the game remains the same.

**Reply by fables-tales** [reply]
> Is that just saying that your snake body acts as if it's continuous through the portal, as opposed to forming a new "tail" after a few moves?

yes, the body is continuous through the portal, so every piece has to go through before the body becomes continuous again
